examples/scripts/client.py

```diff
@@ -0,0 +1,258 @@
+        depends_on: Optional[str] = None,
+        queue: str = "tensorflow_cpu",
+        timeout: str = "24h",
+    ) -> Dict[str, Any]:
+        job_form = {
+            "experiment_name": experiment_name,
+            "queue": queue,
+            "timeout": timeout,
+            "entry_point": entry_point,
+        }
+
+        if entry_point_kwargs is not None:
+            job_form["entry_point_kwargs"] = entry_point_kwargs
+
+        if depends_on is not None:
+            job_form["depends_on"] = depends_on
+
+        workflows_file = Path(workflows_file)
+
+        with workflows_file.open("rb") as f:
+            job_files = {"workflow": (workflows_file.name, f)}
+            response = requests.post(
+                self.job_endpoint,
+                data=job_form,
+                files=job_files,
+            )
+
+        return response.json()
+
+    def upload_custom_plugin_package(
+        self,
+        custom_plugin_name: str,
+        custom_plugin_file: PathLike,
+        collection: str = "dioptra_custom",
+    ) -> Dict[str, Any]:
+        plugin_upload_form = {
+            "task_plugin_name": custom_plugin_name,
+            "collection": collection,
+        }
+
+        custom_plugin_file = Path(custom_plugin_file)
+
+        with custom_plugin_file.open("rb") as f:
+            custom_plugin_file = {"task_plugin_file": (custom_plugin_file.name, f)}
+            response = requests.post(
+                self.task_plugin_endpoint,
+                data=plugin_upload_form,
+                files=custom_plugin_file,
+            )
+
+        return response.json()
+
+
+def tar_helper(
+    source_dir: list[Union[str, Path]],
+    tarball_filename: str,
+    tarball_write_mode: str = "w:gz",
+    working_dir: Optional[Union[str, Path]] = None,
+) -> None:
+    """Archives a list of directories and files and compresses the archive.
+    Args:
+        source_dir: The directories and files which should be archived.
+        tarball_filename: The filename to use for the archived tarball.
+        tarball_write_mode: The write mode for the tarball, see :py:func:`tarfile.open`
+            for the full list of compression options. The default is `"w:gz"` (gzip
+            compression).
+        working_dir: The location where the file should be saved. If `None`, then the
+            current working directory is used. The default is `None`.
+    Returns:
+        Path: Path of the archive.
+    See Also:
+        - :py:func:`tarfile.open`
+    """
+    if working_dir is None:
+        working_dir = Path.cwd()
+
+    working_dir = Path(working_dir)
+    tarball_path = working_dir / tarball_filename
+
+    with tarfile.open(tarball_path, tarball_write_mode) as f:
+        for dir in source_dir:
+            dir = Path(dir)
+            f.add(dir, arcname=dir.name)
+
+    # uses octal so 436 in octal is 664, could alternatively use 0o to represent octal numbers.
+    os.chmod(tarball_path, 436)
```

Comment:
I don't think `os.chmod` is necessary here, try removing it and confirm that nothing is broken.